kingpin/actors/aws/base.py

```diff
@@ -83,19 +85,34 @@ def __init__(self, *args, **kwargs):
 
         super(AWSBaseActor, self).__init__(*args, **kwargs)
 
-        if not (aws_settings.AWS_ACCESS_KEY_ID and
+        # By default, we will try to let Boto handle discovering its
+        # credentials at instantiation time. This _can_ result in synchronous
+        # API calls to the Metadata service, but those should be fast.
+        key = None
+        secret = None
+
+        # In the event though that someone has explicitly set the AWS access
+        # keys in the environment (either for the purposes of a unit test, or
+        # because they wanted to), we use those values.
+        if (aws_settings.AWS_ACCESS_KEY_ID and
```

Comment:
The only reason I put in the option for overriding it was for the unit tests -- its easier to do the forcing of the settings in the aws.settings module and then reloading that, than actually mocking everything out. Maybe there is a more clever way to globally mock out the aws settings so that we know for sure that during a unit test we're not reaching out to an API.